src/rust/engine/rule_graph/src/lib.rs

```diff
@@ -1008,13 +1069,28 @@ impl<R: Rule> RuleGraph<R> {
       .filter_map(|(k, deps)| match k {
         EntryWithDeps::Root(_) => {
           let root_str = entry_with_deps_str(k);
+          let mut dep_entries = deps
+            .all_dependencies()
+            .map(|d| entry_node_str_with_attrs(d))
+            .collect::<Vec<_>>();
+          dep_entries.sort();
+          let deps_with_attrs = dep_entries
+            .iter()
+            .cloned()
```

Comment:
Do we need this `cloned()`? This code `map`s the iterator of `Rule`-implementing types to `String` it looks like, and the string format should be fine with not moving the values of this iterator.

Reply:
It looks like the answer is no, but this `visualize()` method I think should be called at most once per pants execution? I have made this change though!